### ISSUE TITLE
Pixelfed and Mastodon on same server - 503 error when searching for local Pixelfed account from local Mastodon

### ISSUE THREAD

**Issue by chrisale** [pixelfed/pixelfed#4657]
Hi all,

(I've created a corresponding bug report on the Mastodon GitHub as well at
https://github.com/mastodon/mastodon/issues/26962
)

I have Pixelfed and Mastodon running natively together, in separate user accounts, on a Debian 11 server.  All appears to be working, people can search for, discover and follow @chris@pixelfed.chrisalemany.ca from any software on the Fediverse and I can interact normally from Pixelfed. Likewise, my Mastodon account is fully functional.

The *exception* is the connection between my local Pixelfed and Mastodon accounts.

1) When I search for @chris@pixelfed.chrisalemany.ca from the mastodon web client at mstdn.chrisalemany.ca, Mastodon fails and reports: 

`503Remote data could not be fetched`

nginx/access.log logs the following:

`[18/Sep/2023:08:35:59 -0700] "GET /api/v2/search?q=%40chris%40pixelfed.chrisalemany.ca&resolve=true&limit=5 HTTP/2.0" 503 44 "https://mstdn.chrisalemany.ca/getting-started" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.6 Safari/605.1.15"
`
Nothing is produced in the nginx/error log.

I am running Mastodon 4.1.7 and Pixelfed 0.11.9
Enabling ElasticSearch on Mastodon has no effect. (I generally don't have it enabled)

-------

2) The second issue is likely related to the first, so I will only put it up here for added context.  I *can* see and follow my mastodon account @chris@mstdn.chrisalemany from @chris@pixelfed.chrisalemany.ca. However, new posts from my Mastodon do not actually appear in my home Pixelfed timeline.  Again, I suspect this is just another symptom of the same issue as causes the first problem with following from Mastodon.

**Comment by x64x2** [comment]
Have you tried restarting server??

**Comment by chrisale** [comment]
yes. the Mastodon folks think it might be related to my running the servers inside a NAT network at my home which necessitated me specifying the pixelfed domain on the local /etc/hosts file. This is apparently unsupported on Mastodon for security reasons.
